dask/dataframe/reshape.py

```diff
@@ -161,19 +163,33 @@ def get_dummies(
 
     # We explicitly create `meta` on `data._meta` (the empty version) to
     # work around https://github.com/pandas-dev/pandas/issues/21993
-    meta = pd.get_dummies(
-        data._meta,
-        prefix=prefix,
-        prefix_sep=prefix_sep,
-        dummy_na=dummy_na,
-        columns=columns,
-        sparse=sparse,
-        drop_first=drop_first,
-        **kwargs
-    )
+    if isinstance(data._meta, (pd.Series, pd.Index, pd.DataFrame)):
+        meta = pd.get_dummies(
+            data._meta,
+            prefix=prefix,
+            prefix_sep=prefix_sep,
+            dummy_na=dummy_na,
+            columns=columns,
+            sparse=sparse,
+            drop_first=drop_first,
+            **kwargs
+        )
+        method_cache = pd
```

Comment:
I don't understand the use of the term "method cache" for this.

Reply:
This should go away if I have a better way to call get_dummies rather than from Method cache. But my understanding was that M was a Method cache for package functions. But I'm wrong and it doesn't seem to be the case.